File: app/src/main/java/android/gpuimage/com/gpuimageandroid/MainActivity.java
package android.gpuimage.com.gpuimageandroid;

import android.gpuimage.com.gpuimage.GPUImageGaussianBlurFilter;
import android.gpuimage.com.gpuimage.GPUImageView;
import android.graphics.Bitmap;
import android.graphics.BitmapFactory;
import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        Bitmap bmp = BitmapFactory.decodeResource(getResources(), R.drawable.image);
        GPUImageView gpuimage_view = findViewById(R.id.gpuimage_view);
        gpuimage_view.setImage(bmp);
        gpuimage_view.setFilter(new GPUImageGaussianBlurFilter(5.0f));
        gpuimage_view.requestRender();

    }
}

File: app/src/main/java/android/gpuimage/com/gpuimageandroid/MainActivity.kt
package android.gpuimage.com.gpuimageandroid

import android.graphics.Bitmap
import android.os.Bundle
import android.os.Handler
import android.sticky.com.ui.icon.IconStickyFragment
import android.support.v7.app.AppCompatActivity
import android.view.Window
import android.view.WindowManager
import com.sample.huutho.utils.decodeResource
import com.sample.huutho.utils.getBitmapFromTempCache
import com.sample.huutho.utils.saveBitmapToTempCache
import kotlinx.android.synthetic.main.activity_main.*

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        requestWindowFeature(Window.FEATURE_NO_TITLE);
        window.setFlags(WindowManager.LayoutParams.FLAG_FULLSCREEN,WindowManager.LayoutParams.FLAG_FULLSCREEN);
        setContentView(R.layout.activity_main)

        decodeResource(
                this,
                R.drawable.girl,
                start = {},
                success = { bmp: Bitmap ->
                    saveBitmapToTempCache(
                            bmp,
                            { supportFragmentManager
                                    .beginTransaction()
                                    .addToBackStack("AdjustFragment")
                                    .add(R.id.root, IconStickyFragment.newInstance())
                                    .commit()},
                            {})
                },
                failure = {})


        supportFragmentManager.addOnBackStackChangedListener {

            Handler().postDelayed({
                getBitmapFromTempCache({},{bmp : Bitmap -> image_view.setImageBitmap(bmp)},{})
            },5000)


        }

    }

}
